clang, Linux 6.7.12, zfs 2.2.7: build error

<!-- Please fill out the following template, which will help other contributors address your issue. -->

<!--
Thank you for reporting an issue.

*IMPORTANT* - Please check our issue tracker before opening a new issue.
Additional valuable information can be found in the OpenZFS documentation
and mailing list archives.

Please fill in as much of the template as possible.
-->

### System information
<!--  add version after "|" character -->
Type | Version/Name
 --- | ---
Distribution Name	| Debian
Distribution Version	| unstable
Kernel Version	| 6.7.12
Architecture	| amd64
OpenZFS Version	| 2.2.7
<!--
Command to find OpenZFS version:
zfs version

Commands to find kernel version:
uname -r		# Linux
freebsd-version -r	# FreeBSD
-->

### Describe the problem you're observing

Trying to compile with clang results in:

```
  CC [M]  /var/lib/dkms/zfs/2.2.7/build/module/lua/lcorolib.o
In file included from /var/lib/dkms/zfs/2.2.7/build/module/icp/algs/blake3/blake3_impl.c:284:
In file included from /var/lib/dkms/zfs/2.2.7/build/module/icp/include/generic_impl.c:37:
In file included from /var/lib/dkms/zfs/2.2.7/build/include/sys/zio_checksum.h:31:
In file included from /var/lib/dkms/zfs/2.2.7/build/include/sys/zio.h:39:
In file included from /var/lib/dkms/zfs/2.2.7/build/include/sys/spa.h:46:
In file included from /var/lib/dkms/zfs/2.2.7/build/include/sys/dmu.h:50:
In file included from /var/lib/dkms/zfs/2.2.7/build/include/sys/zio_compress.h:33:
/var/lib/dkms/zfs/2.2.7/build/include/sys/abd.h:68:3: error: redefinition of typedef 'abd_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
   68 | } abd_t;
      |   ^
/var/lib/dkms/zfs/2.2.7/build/include/os/linux/zfs/sys/abd_os.h:47:20: note: previous definition is here
   47 | typedef struct abd abd_t;
      |                    ^
  CC [M]  /var/lib/dkms/zfs/2.2.7/build/module/lua/lctype.o
  CC [M]  /var/lib/dkms/zfs/2.2.7/build/module/lua/ldebug.o
1 error generated.
make[5]: *** [scripts/Makefile.build:243: /var/lib/dkms/zfs/2.2.7/build/module/icp/algs/blake3/blake3_impl.o] Error 1
make[5]: *** Waiting for unfinished jobs....

```

### Describe how to reproduce the problem

Happens every time. If needed, I can send the kernel config as well.

### Include any warning/errors/backtraces from the system logs
<!--
*IMPORTANT* - Please mark logs and text output from terminal commands
or else Github will not display them correctly.
An example is provided below.

Example:
```
this is an example how log text should be marked (wrap it with ```)
```
-->

--- a/include/os/linux/spl/sys/taskq.h
+++ b/include/os/linux/spl/sys/taskq.h
@@ -33,6 +33,7 @@
 #include <sys/thread.h>
 #include <sys/rwlock.h>
 #include <sys/wait.h>
+#include <sys/kstat.h>
 
 #define	TASKQ_NAMELEN		31
 

--- a/include/os/linux/zfs/sys/abd_os.h
+++ b/include/os/linux/zfs/sys/abd_os.h
@@ -30,6 +30,8 @@
 extern "C" {
 #endif
 
+struct abd;
+
 struct abd_scatter {
 	uint_t		abd_offset;
 	uint_t		abd_nents;
@@ -41,10 +43,8 @@ struct abd_linear {
 	struct scatterlist *abd_sgl; /* for LINEAR_PAGE */
 };
 
-typedef struct abd abd_t;
-
 typedef int abd_iter_page_func_t(struct page *, size_t, size_t, void *);
-int abd_iterate_page_func(abd_t *, size_t, size_t, abd_iter_page_func_t *,
+int abd_iterate_page_func(struct abd *, size_t, size_t, abd_iter_page_func_t *,
     void *);
 
 /*
@@ -52,8 +52,8 @@ int abd_iterate_page_func(abd_t *, size_t, size_t, abd_iter_page_func_t *,
  * Note: these are only needed to support vdev_classic. See comment in
  * vdev_disk.c.
  */
-unsigned int abd_bio_map_off(struct bio *, abd_t *, unsigned int, size_t);
-unsigned long abd_nr_pages_off(abd_t *, unsigned int, size_t);
+unsigned int abd_bio_map_off(struct bio *, struct abd *, unsigned int, size_t);
+unsigned long abd_nr_pages_off(struct abd *, unsigned int, size_t);
 
 #ifdef __cplusplus
 }

